### ISSUE TITLE
Longer labels should be truncated

### ISSUE THREAD

**Issue by theiliad** [carbon-design-system/carbon-charts#347]
![image](https://user-images.githubusercontent.com/14989804/65822141-1ff34680-e20d-11e9-9fcf-d9311b1ea7f3.png)

![image](https://user-images.githubusercontent.com/14989804/76541796-7e7a4d00-645a-11ea-84f7-36589e1c502b.png)

Labels that do not fit into the container should be truncated. We will show a tooltip with the full label text onHover.

#287

**Comment by pcoccoli** [comment]
I definitely need this capability for x-axis labels on bar charts.  Having it in Q4 would be awesome.

**Comment by kalai001** [comment]
Hi @theiliad , @ninja511 , the resolution is getting postponed. Note that this is a regression from the earlier implementation. We shouldn't be keeping regression open for this long. At least bring this to an earlier working level, where it had the below capabilities

1. Labels rendered with `+ 45` degree inclination (Now it is at `-45`, look awkward)
2. Long labels truncates gracefully. (Now it squeezes the graph area, leaving the chart glaringly distorted)

![image](https://user-images.githubusercontent.com/46040341/82890521-f2ac9300-9f69-11ea-8daf-eb42ffcb92b7.png)

**Comment by theiliad** [comment]
Closed by #646
